config/account.go

```diff
@@ -297,3 +303,36 @@ func (a *AccountChannel) IsSet() bool {
 type AccountPrivacy struct {
 	AllowActivities AllowActivities `mapstructure:"allowactivities" json:"allowactivities"`
 }
+
+type IpMasking struct {
+	IpV6 IpMasks `mapstructure:"ipv6" json:"ipv6"`
+	IpV4 IpMasks `mapstructure:"ipv4" json:"ipv4"`
+}
+
+type IpMasks struct {
+	ActivityLeftMaskBits    int `mapstructure:"activity_left_mask_bits" json:"activity_left_mask_bits"`
+	GdprLeftMaskBitsLowest  int `mapstructure:"gdpr_left_mask_bits_lowest" json:"gdpr_left_mask_bits_lowest"`
+	GdprLeftMaskBitsHighest int `mapstructure:"gdpr_left_mask_bits_highest" json:"gdpr_left_mask_bits_highest"`
+}
+
+func (ipMasking *IpMasking) Validate(errs []error) []error {
+	errs = ipMasking.IpV6.validate(errs, Ipv6Bits, "ipv6")
+	errs = ipMasking.IpV4.validate(errs, Ipv4Bits, "ipv4")
+	return errs
+}
+
+func (ipMasks *IpMasks) validate(errs []error, maxBits int, ipVersion string) []error {
+	if ipMasks.ActivityLeftMaskBits > maxBits || ipMasks.ActivityLeftMaskBits < 0 {
+		err := fmt.Errorf("activity left mask bits cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
+		errs = append(errs, err)
+	}
+	if ipMasks.GdprLeftMaskBitsLowest > maxBits || ipMasks.GdprLeftMaskBitsLowest < 0 {
+		err := fmt.Errorf("gdpr left mask bits lowest cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
+		errs = append(errs, err)
+	}
+	if ipMasks.GdprLeftMaskBitsHighest > maxBits || ipMasks.GdprLeftMaskBitsHighest < 0 {
+		err := fmt.Errorf("gdpr left mask bits highest cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
+		errs = append(errs, err)
+	}
+	return errs
+}
```

Comment:
I recommend to break the errors array pattern here. It forces each validate method to understand too much context from it's caller. Instead, consider returning an array and appending as necessary. Something like:

```
func (ipMasks *IpMasks) validate(maxBits int, ipVersion string) []error {
	var errs []error

	if ipMasks.ActivityLeftMaskBits > maxBits || ipMasks.ActivityLeftMaskBits < 0 {
		err := fmt.Errorf("activity left mask bits cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
		errs = append(errs, err)
	}

	if ipMasks.GdprLeftMaskBitsLowest > maxBits || ipMasks.GdprLeftMaskBitsLowest < 0 {
		err := fmt.Errorf("gdpr left mask bits lowest cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
		errs = append(errs, err)
	}

	if ipMasks.GdprLeftMaskBitsHighest > maxBits || ipMasks.GdprLeftMaskBitsHighest < 0 {
		err := fmt.Errorf("gdpr left mask bits highest cannot exceed %d in %s address, or be less than 0", maxBits, ipVersion)
		errs = append(errs, err)
	}

	return errs
}

func (ipMasking *IpMasking) Validate(errs []error) []error {
        var errs []error

	if e := ipMasking.IpV6.validate(errs, Ipv6Bits, "ipv6"); e != nil {
                errs = append(errs, e)
        }
        
	if errs = ipMasking.IpV4.validate(errs, Ipv4Bits, "ipv4") ; e != nil {
                errs = append(errs, e)
        }

	return errs
}
```

Reply:
Modified